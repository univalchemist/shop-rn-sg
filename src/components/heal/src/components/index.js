import { StackBackButton, healNavigationOptions } from './NavigationButtons';
import NonTouchableSearchBar from './NonTouchableSearchBar';
import DoctorCard from './DoctorCard';
import ClinicInfo from './ClinicInfo';
export { default as ErrorView } from './ErrorView';
export { default as SpinningLoader } from './SpinningLoader';
export { default as RadioBoxes } from './RadioBoxes';
export { default as RadioBoxesWithForm } from './RadioBoxesWithForm';
export {
  DoctorCard,
  StackBackButton,
  healNavigationOptions,
  NonTouchableSearchBar,
  ClinicInfo,
};
