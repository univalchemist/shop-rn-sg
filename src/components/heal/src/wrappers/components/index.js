import React from 'react';

export {
  Image,
  Text,
  PlainText,
  Box,
  Button,
  Divider,
  Flex,
  SectionHeadingText,
  ErrorPanel,
  ListItemWithRightChevron,
  ScrollView,
  Icon,
  Loader,
  StatusPanel,
} from '@cxa-rn/components';
export {
  withTracking,
  TrackedButton,
  TrackedFloatingActionButton,
  TrackedListItem,
  TrackedCarousel,
  TrackedCarouselWithScrollBar,
} from './Tracking';
