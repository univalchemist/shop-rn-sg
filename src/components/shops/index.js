export { ShopNavigator } from './src/navigation/Navigator';
export { default as zh_HK } from './src/locales/zh-HK.json';
export { default as en_HK } from './src/locales/en-HK.json';
export {
  initializeShop,
  updateAccountConfigForShop,
  logOutShop,
} from '@shops/store/actions';
export { ShopContextProvider } from './src/ShopProvider';
