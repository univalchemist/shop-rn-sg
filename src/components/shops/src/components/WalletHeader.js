import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import {
  PlainText,
  Box,
  Image,
  TextSkeletonPlaceholder,
} from '@shops/wrappers/components';
import { useIntl, useTheme } from '@shops/wrappers/core/hooks';
import { walletIcon } from '@shops/assets/images';
import { getShopWallet } from '@shops/store/actions';
import { useGetFormattedPrice } from '@shops/wrappers/core/hooks';

const WalletHeader = ({ getShopWallet, wallet, isCXA1Integration }) => {
  const intl = useIntl();
  const theme = useTheme();
  const [isLoadingWallet, setIsLoadingWallet] = useState(true);
  const formattedPrice = useGetFormattedPrice(wallet?.balance);

  useEffect(() => {
    const getWallet = async () => {
      try {
        setIsLoadingWallet(true);
        await getShopWallet();
      } catch (e) {
      } finally {
        setIsLoadingWallet(false);
      }
    };

    getWallet();
  }, []);

  if (!isCXA1Integration) {
    return null;
  }

  return (
    <Box
      backgroundColor={theme.colors.white}
      height={54}
      alignItems="center"
      justifyContent="space-between"
      px={32}
      flexDirection="row"
      borderBottomWidth={1}
      borderBottomColor={theme.colors.walletBorder}
    >
      <Box flexDirection="row" alignItems="center">
        <Image source={walletIcon} mr={8} />
        <PlainText>
          {intl.formatMessage({ id: 'shop.common.wallet' })}
        </PlainText>
      </Box>
      {isLoadingWallet ? (
        <TextSkeletonPlaceholder fontSize={16} lineHeight={22} width={100} />
      ) : (
        <PlainText fontWeight="bold">{formattedPrice}</PlainText>
      )}
    </Box>
  );
};

export default connect(
  ({
    user: { isCXA1Integration },
    shop: {
      home: { wallet },
    },
  }) => ({
    wallet,
    isCXA1Integration,
  }),
  {
    getShopWallet,
  },
)(WalletHeader);
