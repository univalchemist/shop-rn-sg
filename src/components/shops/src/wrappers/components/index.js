/* istanbul ignore file */
import React from 'react';
import { useTheme } from '@shops/wrappers/core/hooks';

export {
  Box,
  TouchableContainer,
  Image,
  Card,
  ImageErrorCard,
  ListItem,
  Icon,
  Flex,
  Footer,
  ScrollView,
  Divider,
  Button,
  Input,
  CustomSelectField,
  Carousel,
  TextSkeletonPlaceholder,
} from '@cxa-rn/components';

import {
  Text as TextDefault,
  SectionHeadingText as SectionHeadingTextDefault,
  PlainText as PlainTextDefault,
  SecondaryText as SecondaryTextDefault,
  StackBackButton as StackBackButtonDefault,
} from '@cxa-rn/components';
import { TouchableOpacity } from 'react-native';

export { TrackedButton, ModalBackButton } from '@wrappers/components';
export { InputField } from '@wrappers/components/FinalForm';
export { CustomMultiselectCheckBox } from '@wrappers/components/form';

export const Text = props => {
  const theme = useTheme();

  return <TextDefault {...props} color={props.color || theme.colors.text} />;
};

export const StackBackButton = ({
  onPress,
  icName = 'keyboard-arrow-left',
  ...props
}) => {
  return (
    <TouchableOpacity activeOpacity={1} onPress={onPress} {...props}>
      <StackBackButtonDefault icName={icName} />
    </TouchableOpacity>
  );
};

export const SectionHeadingText = props => {
  const theme = useTheme();

  return (
    <SectionHeadingTextDefault
      {...props}
      color={props.color || theme.colors.text}
    />
  );
};

export const PlainText = props => {
  const theme = useTheme();

  return (
    <PlainTextDefault {...props} color={props.color || theme.colors.text} />
  );
};

export const SecondaryText = props => {
  const theme = useTheme();

  return (
    <SecondaryTextDefault
      {...props}
      color={props.color || theme.colors.label}
    />
  );
};

export { default as TextArea } from './TextArea';
export { default as TextAreaField } from './TextAreaField';
export { default as CheckBox } from './CheckBox';
