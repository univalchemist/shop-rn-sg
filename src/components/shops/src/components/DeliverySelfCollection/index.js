import { Box, CustomSelectField, Icon, Text } from '@shops/wrappers/components';
import { RadioButton, RadioGroup } from '@shops/components';
import { DELIVERY_TYPE } from '@shops/utils/constant';
import { SELECT_ADDRESS_MODAL } from '@shops/navigation/routes';
import React from 'react';
import {
  useGetFormattedPrice,
  useIntl,
  useTheme,
} from '@shops/wrappers/core/hooks';

const DeliverySelfCollection = ({
  product,
  deliveryType,
  setDeliveryType,
  currentSelected,
  onSelectLocation,
  error,
  navigation,
}) => {
  const theme = useTheme();
  const intl = useIntl();
  const deliveryFeeFormatted = useGetFormattedPrice(product.deliveryFee);
  return (
    <Box>
      <Box>
        <Text mb={24}>
          {intl.formatMessage({
            id: 'shop.product.deliveryType',
            defaultMessage: 'Delivery or Self-collection',
          })}
        </Text>
        {product.deliveryMethod && (
          <RadioGroup value={deliveryType} onChange={setDeliveryType}>
            {product.deliveryMethod?.includes(DELIVERY_TYPE.DELIVERY) ? (
              <RadioButton
                text={deliveryFeeFormatted}
                value={DELIVERY_TYPE.DELIVERY}
              />
            ) : (
              <></>
            )}
            {product.deliveryMethod?.includes(DELIVERY_TYPE.SELF_COLLECTION) ? (
              <RadioButton
                text={intl.formatMessage({
                  id: 'shop.product.selfCollection',
                })}
                value={DELIVERY_TYPE.SELF_COLLECTION}
              />
            ) : (
              <></>
            )}
          </RadioGroup>
        )}
      </Box>
      {deliveryType === DELIVERY_TYPE.SELF_COLLECTION && (
        <CustomSelectField
          labelEmptyComponent={
            <Text
              fontSize={18}
              ellipsizeMode="tail"
              numberOfLines={1}
              color={theme.text}
            >
              {intl.formatMessage({
                id: 'shop.product.selectLocation',
              })}
            </Text>
          }
          label={intl.formatMessage({
            id: 'shop.product.selectLocation',
          })}
          input={{
            value: currentSelected?.redemptValue,
          }}
          intl={intl}
          theme={{
            ...theme,
            colors: { ...theme.colors, gray: ['gray'] },
            inputField: { inputBorder: 'black' },
          }}
          meta={{ error, touched: true }}
          onPress={() =>
            navigation.navigate(SELECT_ADDRESS_MODAL, {
              onSelectLocation,
              currentSelected,
              redemptionPoint: product.redemptionPoint,
            })
          }
          onRight={({ color }) => <Icon name="expand-more" color={color} />}
        />
      )}
    </Box>
  );
};

export default DeliverySelfCollection;
