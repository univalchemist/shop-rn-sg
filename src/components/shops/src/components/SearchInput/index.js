/* istanbul ignore file */
//TODO update unit test
import React, { useState, useCallback } from 'react';
import { TextInput, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { Box } from '@shops/wrappers/components';
import { useTheme } from '@shops/wrappers/core/hooks';
import { NAVIGATION_HOME_SCREEN_PATH } from '@shops/navigation/routes';
import { useNavigation } from '@react-navigation/native';

const SearchInput = ({ value, placeholder, onChangeText, onSubmit }) => {
  const theme = useTheme();
  const navigation = useNavigation();
  const [focus, setFocus] = useState(false);

  const setFocusStatus = useCallback(() => {
    setFocus(!focus);
  }, [focus]);

  const onSubmitEditing = useCallback(() => {
    if (value?.length >= 3 || value?.length === 0) {
      onSubmit();
      navigation.navigate(NAVIGATION_HOME_SCREEN_PATH);
    }
  }, [value, onSubmit, navigation]);

  return (
    <Box
      width="100%"
      height={56}
      borderRadius={4}
      borderWidth={0.5}
      pl={16}
      borderColor={focus ? theme.colors.primary[1] : theme.colors.border}
      backgroundColor={focus ? theme.colors.white : theme.colors.primary[1]}
      justifyContent="center"
    >
      <TextInput
        value={value}
        onChangeText={value => onChangeText(value)}
        placeholder={placeholder}
        placeholderTextColor={focus ? theme.colors.text : theme.colors.white}
        style={[
          styles.input,
          { color: focus ? theme.colors.text : theme.colors.white },
        ]}
        autoCapitalize={'none'}
        onBlur={setFocusStatus}
        onFocus={setFocusStatus}
        returnKeyType="search"
        onSubmitEditing={onSubmitEditing}
      />
    </Box>
  );
};
const styles = StyleSheet.create({
  input: {
    fontSize: 16,
  },
});

SearchInput.propTypes = {
  style: PropTypes.object,
  placeholder: PropTypes.string,
  value: PropTypes.string,
  textStyle: PropTypes.object,
  onChangeText: PropTypes.func,
  onSubmit: PropTypes.func,
};

export default SearchInput;
