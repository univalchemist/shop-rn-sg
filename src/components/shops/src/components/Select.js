import { useTheme } from '@shops/wrappers/core/hooks';
import { StyleSheet, TouchableOpacity } from 'react-native';
import { Icon, Box } from '@shops/wrappers/components';
import React from 'react';

const Select = ({ children, onPress, disabled = false }) => {
  const theme = useTheme();

  return (
    <TouchableOpacity
      disabled={disabled}
      style={[styles.selectContainer, { borderColor: theme.colors.border }]}
      activeOpacity={0.5}
      onPress={onPress}
    >
      <Box flex={1}>{children}</Box>

      <Icon name="expand-more" size={30} />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  selectContainer: {
    height: 56,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderRadius: 4,
    borderWidth: 1,
    width: '100%',
    paddingHorizontal: 16,
    flex: 1,
  },
  padding: { marginHorizontal: 20 },
});

export default Select;
