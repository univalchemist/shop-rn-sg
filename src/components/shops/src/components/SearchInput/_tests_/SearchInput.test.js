
import React from 'react';
import { SearchInput } from '@shops/components';
import { renderForTest } from '@testUtils';

const initialState = {};

describe('SearchInput', () => {
  test('should match snapshot', async () => {
    const value = '';
    const placeholder = 'Search';
    const onChangeText = jest.fn();
    const onSubmit = jest.fn();
    const Component = renderForTest(
      <SearchInput
        value={value}
        placeholder={placeholder}
        onChangeText={onChangeText}
        onSubmit={onSubmit}
      />,
      {
        initialState,
      },
    );
    expect(Component.toJSON()).toMatchSnapshot();
  });
});
