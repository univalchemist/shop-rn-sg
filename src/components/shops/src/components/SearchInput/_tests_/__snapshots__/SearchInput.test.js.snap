// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        backgroundColor="#212121"
                        borderColor="#9E9E9E"
                        borderRadius={4}
                        borderWidth={0.5}
                        height={56}
                        justifyContent="center"
                        pl={16}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#212121",
                              "borderColor": "#9E9E9E",
                              "borderRadius": 4,
                              "borderWidth": 0.5,
                              "height": 56,
                              "justifyContent": "center",
                              "paddingLeft": 16,
                              "width": "100%",
                            },
                          ]
                        }
                        width="100%"
                      >
                        <TextInput
                          allowFontScaling={true}
                          autoCapitalize="none"
                          onBlur={[Function]}
                          onChangeText={[Function]}
                          onFocus={[Function]}
                          onSubmitEditing={[Function]}
                          placeholder="Search"
                          placeholderTextColor="#fff"
                          rejectResponderTermination={true}
                          returnKeyType="search"
                          style={
                            Array [
                              Object {
                                "fontSize": 16,
                              },
                              Object {
                                "color": "#fff",
                              },
                            ]
                          }
                          underlineColorAndroid="transparent"
                          value=""
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
